lib/reservation_booker.rb

```diff
@@ -0,0 +1,81 @@
+require_relative "reservation"
+require_relative "all_rooms"
+require_relative "date_range"
+require_relative "room"
+
+module Hotel
+  class ReservationBooker
+    attr_reader :all_reservations
+
+    def initialize
+      @all_rooms = Hotel::AllRooms.new
```

Comment:
It would have been OK to let `ReservationBooker` and `AllRooms` be the same class. The booker would have been responsible for managing the rooms and the reservations, which would create coupling, but to an acceptable degree.